zerver/lib/export.py

```diff
@@ -1182,26 +1189,34 @@ def _get_exported_s3_record(
     if processing_emoji:
         record['file_name'] = os.path.basename(key.name)
 
-    # A few early avatars don't have 'realm_id' on the object; fix their metadata
-    user_profile = get_user_profile_by_id(record['user_profile_id'])
-    if 'realm_id' not in record:
-        record['realm_id'] = user_profile.realm_id
-    record['user_profile_email'] = user_profile.email
+    if "user_profile_id" in record:
+        user_profile = get_user_profile_by_id(record['user_profile_id'])
+        record['user_profile_email'] = user_profile.email
 
-    # Fix the record ids
-    record['user_profile_id'] = int(record['user_profile_id'])
-    record['realm_id'] = int(record['realm_id'])
+        # Fix the record ids
+        record['user_profile_id'] = int(record['user_profile_id'])
 
-    return record
+        # A few early avatars don't have 'realm_id' on the object; fix their metadata
+        if 'realm_id' not in record:
+            record['realm_id'] = user_profile.realm_id
```

Comment:
Can you talk about how you found early avatars without a `realm_id`?  I'm wondering if there's a bug.